python/cudf/cudf/tests/test_series.py

```diff
@@ -1813,3 +1813,17 @@ def test_series_digitize_invalid_bins():
         ValueError, match="`bins` cannot contain null entries."
     ):
         _ = s.digitize(bins)
+
+
+@pytest.mark.parametrize("data", [[0, 1, 2, 3, 4, 5, 10]])
+@pytest.mark.parametrize("left", [0, 2, 5])
+@pytest.mark.parametrize("right", [10, 1, 4])
+@pytest.mark.parametrize("inclusive", ["both", "neither", "left", "right"])
+def test_series_between(data, left, right, inclusive):
```

Comment:
We need to test list-like inputs as well. I would test `list`, `pd.Series`, and `cudf.Series`. Example of desired behavior:

```python
>>> cudf.Series(range(10)).between([5,]*10, [8,]*10)
0    False
1    False
2    False
3    False
4    False
5     True
6     True
7     True
8     True
9    False
dtype: bool
```

Note that this example computes a comparison between a GPU object `cudf.Series` and host object `list`. I know we've discussed disabling binary ops between GPU and host objects. I'm not 100% sure how we want to proceed here. To avoid computing `self > left` and similar expressions on a host object, I think you might need to pre-process `left, right` to convert host list-like objects into a `cudf.Series` before computing the comparator binary ops.

Reply:
Done 👍